packages/cli-lib/lib/uploadFolder.js

```diff
@@ -28,32 +30,82 @@ const queue = new PQueue({
   concurrency: 10,
 });
 
-function getFilesByType(files) {
+function getFilesByType(files, src) {
   const moduleFiles = [];
   const cssAndJsFiles = [];
   const otherFiles = [];
   const templateFiles = [];
   const jsonFiles = [];
+  const compiledJsonFiles = [];
+
+  const fieldsJsInRoot = listFilesInDir(src).includes('fields.js');
 
   files.forEach(file => {
     const parts = splitLocalPath(file);
     const extension = getExt(file);
-
     const moduleFolder = parts.find(part => part.endsWith('.module'));
+    const fileName = parts[parts.length - 1];
+    const options = yargs.argv.options;
     if (moduleFolder) {
-      moduleFiles.push(file);
+      //If the folder contains a fields.js, we will always overwrite the existing fields.json.
```

Comment:
I think that we should avoid overwriting or deleting any customer code here. I think we could either ignore `fields.json` when `fields.js` is present or prompt the user for the correct file (or maybe to overwrite?)

Reply:
Because we write a file called `fields.json` when we compile the `fields.js`, I don't think we could ignore it. Perhaps we could rename it to `fields.original.json` or something. I do like the idea of prompting the user as well.

Reply:
For some other [build-artifacty commands](https://github.com/HubSpot/hubspot-cli/blob/7de0079d5054b8102a16ffa1698e29e27e0d83bf/packages/cli-lib/archive.js#L28) we've written files to a `tmp` directory and I'm wondering if that's the safer bet for this too. That way we can avoid making any unexpected changes in/around the customer's source files

Reply:
Looks like a solid plan!

Just in case you haven't seen them, lots of prompt examples here https://github.com/HubSpot/hubspot-cli/tree/817283def17783d01df3b69ce159c87321add4e9/packages/cli/lib/prompts

Reply:
After talking w/ Anthony, decided to push prompting off for now, but have add tmp folder writing, as well as a `--saveOutput` command that will write any built output to a `fields.output.json` file in the same folder as the `fields.js`.